rustls/src/error.rs

```diff
@@ -60,17 +60,20 @@ pub enum Error {
     /// We received a fatal alert.  This means the peer is unhappy.
     AlertReceived(AlertDescription),
 
-    /// We received an invalidly encoded certificate from the peer.
-    InvalidCertificateEncoding,
-
-    /// We received a certificate with invalid signature type.
-    InvalidCertificateSignatureType,
-
-    /// We received a certificate with invalid signature.
-    InvalidCertificateSignature,
-
-    /// We received a certificate which includes invalid data.
-    InvalidCertificateData(String),
+    /// We saw an invalid certificate.
+    ///
+    /// This includes, but is not limited to:
+    /// - we received an invalidly encoded certificate,
+    /// - we received a certificate that was not signed correctly
+    ///   by its alleged issuer,
+    /// - we received a certificate that was not issued by any
+    ///   of the configured trusted root certificates,
+    /// - we received a certificate that was not valid for the expect
+    ///   server name.
```

Comment:
I probably wouldn't keep this list here, just refer to the docs for `CertificateError`?

Reply:
yeah will do that!